test/e2e/aks_versions.go

```diff
@@ -106,21 +105,23 @@ func GetWorkingAKSKubernetesVersion(ctx context.Context, subscriptionID, locatio
 	baseVersion := fmt.Sprintf("%s.0", semver.MajorMinor(version))
 	maxVersion := fmt.Sprintf("%s.0", semver.MajorMinor(version))
 	var foundWorkingVersion bool
-	for _, o := range *result.Orchestrators {
-		orchVersion := *o.OrchestratorVersion
-		// semver comparisons require a "v" prefix
-		if orchVersion[:1] != "v" {
-			orchVersion = fmt.Sprintf("v%s", *o.OrchestratorVersion)
-		}
-		// if the inputted version matches with an available AKS version we can return immediately
-		if orchVersion == version && !latestStableVersionDesired {
-			return version, nil
-		}
-
-		// or, keep track of the highest aks version for a given major.minor
-		if semver.MajorMinor(orchVersion) == semver.MajorMinor(maxVersion) && semver.Compare(orchVersion, maxVersion) >= 0 {
-			maxVersion = orchVersion
-			foundWorkingVersion = true
+	for _, minor := range result.KubernetesVersionListResult.Values {
```

Comment:
This API changed from SDK v1 to v2, so we have to iterate versions slightly differently.

Reply:
since we already know the minor of the desired AKS minor version do we really need to range over all patches of all minors?

Reply:
Good point, we can do the `MajorMinor` comparison first in the loop and just `continue` if they aren't the same. I'll change that.